### PR TITLE
kicad: add srcs parameter to allow configuring kicad versions

### PR DESCRIPTION
**Pull request by matthuszagh**
###### Motivation for this change
Allows overriding kicad versions which are otherwise inaccessible.

###### Things done
Added srcs parameter that can be used to customize kicad or kicad-unstable versions, including libraries. For instance, you could do something like:
```nix
final: prev:

{
  kicad-unstable = (prev.kicad-unstable.override {
    srcs = {
      "kicad-unstable" = {
        kicadVersion = {
          version = "2020-10-08";
          src = {
            rev = "fd22fe8e374ce71d57e9f683ba996651aa69fa4e";
            sha256 = "sha256-F8qugru/jU3DgZSpQXQhRGNFSk0ybFRkpyWb7HAGBdc=";
          };
        };
        libVersion = {
          version = "2020-08-22";
          libSources = {
            i18n.rev = "cbbb1efd940094bf0c3168280698b2b059a8c509";
            i18n.sha256 = "1q4jakn6m8smnr2mg7jgb520nrb6fag9mdvlcpx3smp3qbxka818";
            symbols.rev = "9ca6a5348cdeb88e699582d4ed051ff7303b44d3";
            symbols.sha256 = "13w6pb34rhz96rnar25z7kiscy6q1fm8l39hq1bpb8g9yn86ssz4";
            templates.rev = "ae16953b81055855bcede4a33305413599d86a15";
            templates.sha256 = "1pkv90p3liy3bj4nklxsvpzh9m56p0k5ldr22armvgqfaqaadx9v";
            footprints.rev = "f94c2d5d619d16033f69a555b449f59604d97865";
            footprints.sha256 = "1g71sk77jvqaf9xvgq6dkyvd9pij2lb4n0bn0dqnwddhwam935db";
            packages3d.rev = "f699b0e3c13fe75618086913e39279c85da14cc7";
            packages3d.sha256 = "0m5rb5axa946v729z35ga84in76y4zpk32qzi0hwqx957zy72hs9";
          };
        };
      };
    };
}
```

I've also reformatted the existing code. Let me know if the reformatting is too invasive and I'll revert.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
@matthuszagh this change is somewhat simple enough so it's easy for me to press the button, but I think it would be nice if it was possible to use a different _fork_ of a library / kicad itself, not only a different revision.

P.S the eval errors have nothing to do with you.

**Comment by matthuszagh** [comment]
I didn't consider that but I agree with you. Let me dig into this a little more.

**Comment by matthuszagh** [comment]
@doronbehar this now allows overriding with a full `fetchFromGitLab`, or whatever you want. Versions can also be overriden with `kicadVersion` or `libVersion`. For example, this is how I'm using it:

```nix
final: prev:

{
  kicad-unstable = (prev.kicad-unstable.override {
    symbol-path = "/home/matt/src/kicad-symbols";
    footprint-path = "/home/matt/src/kicad-footprints";
    model3d-path = "/home/matt/src/kicad-packages3D";
    srcs = {
      kicadVersion = "2020-10-08";
      kicad = prev.fetchFromGitLab {
        group = "kicad";
        owner = "code";
        repo = "kicad";
        rev = "fd22fe8e374ce71d57e9f683ba996651aa69fa4e";
        sha256 = "sha256-F8qugru/jU3DgZSpQXQhRGNFSk0ybFRkpyWb7HAGBdc=";
      };
    };
  });
}
```

Let me know what you think.

File: pkgs/applications/science/electronics/kicad/libraries.nix [hunk 1]
```diff
@@ -1,39 +1,27 @@
...
 
-# callPackage libraries {
-#   version = "unstable";
-#   libs.symbols = {
-#     rev = "09f9..";
-#     sha256 = "...";
-#   };
-# };
-with lib;
+with stdenv.lib;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/libraries.nix:8]: https://nix.dev/anti-patterns/language.html#with-attrset-expression

File: pkgs/applications/science/electronics/kicad/i18n.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ stdenv
+, cmake
+, gettext
+, i18nSrc
+, i18nVersion
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/i18n.nix:5]: Nitpick: Use `src` and `version` here, and `inherit src version` under mkDerivation.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 3]
```diff
@@ -1,52 +1,119 @@
...
+
+  libSrcFetch = name: fetchFromGitLab {
+    group = "kicad";
+    owner = "libraries";
+    repo = "kicad-${name}";
+    rev = versionsImport.${baseName}.libVersion.libSources.${name}.rev;
+    sha256 = versionsImport.${baseName}.libVersion.libSources.${name}.sha256;
+  };
+
+  srcOverridep = attr: (!stable && srcs ? ${attr});
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:64]: More readble IMO:

```suggestion
  srcOverridep = attr: (!stable && builtins.hasAttr attr srcs);
```

**Review by doronbehar — COMMENTED** [review]
Besides these, here are general comments:

1. It'd be easier to review this if the indentation settings will be in a separate commit.
2. More comments to how the logic is going on inside `default.nix` regarding `srcs`.
3. Since there's no way to query a package's override options, it'd be nice to put a big comment at the start of `default.nix` with all the possible ways one can override `srcs`.

**Comment by doronbehar** [comment]
Oh and 1 more thing: I'm rather sure this change shouldn't trigger rebuilds - ofborg detects it [this way](https://github.com/NixOS/ofborg#running-meta-checks-locally) and in the "eval" check of the PR, there's a link to the list of changed outputs. Also the PR's labels ofborg sets reflects that.

**Comment by matthuszagh** [comment]
@doronbehar thanks for the review. I've applied the changes. Let me know.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 4]
```diff
@@ -1,52 +1,165 @@
...
+      };
 
   python = python3;
   wxPython = python.pkgs.wxPython_4_0;
 
 in
 stdenv.mkDerivation rec {
 
-  passthru.libraries = callPackages ./libraries.nix versionConfig.libVersion;
+  passthru.libraries = callPackages ./libraries.nix { inherit libSrc libVersion; };
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:148]:
```suggestion
  # Common libraries, referenced during runtime, via the wrapper.
  passthru.libraries = callPackages ./libraries.nix { inherit libSrc libVersion; };
```

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 5]
```diff
@@ -46,29 +49,22 @@ assert ngspiceSupport -> libngspice != null;
 
 with lib;
 let
-  versionConfig = versions.${baseName};
-
   # oce on aarch64 fails a test
   withOCE = oceSupport && !stdenv.isAarch64;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:53]: This is not part of the PR, I know. This should be an assertion and not a silent disable.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 6]
```diff
@@ -46,29 +49,22 @@ assert ngspiceSupport -> libngspice != null;
...
   withOCC = (withOCCT && !withOCE) || (oceSupport && stdenv.isAarch64);
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:54]: This as well, put the transition to an assertion in a separate PR, please.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 7]
```diff
@@ -21,11 +21,14 @@
...
 , stable ? true
 , baseName ? "kicad"
-, versions ? { }
+, kicadSrc
+, kicadVersion
+, i18nSrc
+, i18nVersion
 , oceSupport ? false
 , opencascade
 , withOCCT ? true
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:34]: Using `*Support` variables and `with*` together is misleading, but it's an API that's already exposed. It could be deprecated rather nicely with an assertion as well (`lib.assertions.assertMsg` I think is good).

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 8]
```diff
@@ -46,29 +49,22 @@ assert ngspiceSupport -> libngspice != null;
 
 with lib;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:50]: While at it, we'd be better of without this.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 9]
```diff
@@ -1,52 +1,165 @@
...
+  passthru.i18n = callPackage ./i18n.nix {
```
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:149]: OTH, I agree that putting `i18n` outside of libraries makes more sense, since it's a requirement of `base.nix` and not the rest, so... maybe make `passthru.libraries.i18n = throw "this attribute has moved to "kicad-unstable.passthru.i18n"`. Or maybe use `lib.warn`.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 10]
```diff
@@ -46,29 +49,22 @@ assert ngspiceSupport -> libngspice != null;
...
-  libraries = callPackages ./libraries.nix versionConfig.libVersion;
-
 in
 stdenv.mkDerivation rec {
 
-  i18n = libraries.i18n;
+  i18n = callPackage ./i18n.nix {
+    src = i18nSrc;
+    version = i18nVersion;
+  };
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:62]: This is called twice eventually - not good. I think it fits more if it's in `default.nix` because it's the entry point of whoever that inspects this derivation.

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 9 above)
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:149]: Called twice - the 1st / 2nd time here.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 11]
```diff
@@ -1,52 +1,165 @@
...
+    src = i18nSrc;
+    version = i18nVersion;
+  };
   base = callPackage ./base.nix {
-    inherit versions stable baseName;
+    inherit stable baseName;
+    inherit kicadSrc kicadVersion;
+    inherit i18nSrc i18nVersion;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:156]:
```suggestion
    inherit (passthru) i18n;
```

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 12]
```diff
@@ -59,7 +172,7 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ makeWrapper ]
     ++ optionals (scriptingSupport)
-      [ python.pkgs.wrapPython ];
+    [ python.pkgs.wrapPython ];
 
   # wrapGAppsHook added the equivalent to ${base}/share
   # though i noticed no difference without it
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:178]: Nitpick:

```suggestion
  # We are emulating wrapGAppsHook, along with other variables to the wrapper
```

**Review by doronbehar — COMMENTED** [review]
The review is a bit messy sorry, but the expression is getting better and better :). Some suggestions are better of in separate commits.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 13]
```diff
@@ -46,29 +49,22 @@ assert ngspiceSupport -> libngspice != null;
...
-  version = "${builtins.substring 0 10 versions.${baseName}.kicadVersion.src.rev}";
-
-  src = fetchFromGitLab (
-    {
-      group = "kicad";
-      owner = "code";
-      repo = "kicad";
-    } // versionConfig.kicadVersion.src
-  );
+  version = kicadVersion;
```
[the remark this answers is not in the document]
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/base.nix:65]: Would you mind clarifying this point? I'm not sure I completely understand...

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/default.nix:149]: I'm not sure I completely understand this point. Why do we need to pass i18n to libraries at all? Aren't libraries only called through kicad?

**Comment by matthuszagh** [comment]
@doronbehar I hope I got everything (I think I did), but sorry in advance if I messed something up (got into a little trouble with all the rebasing :)).

Also let me know about those other two points.

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 12 above)
Comment (evils) [reply pkgs/applications/science/electronics/kicad/default.nix]: i wasn't emulating wrapGAppsHook, since that misbehaves

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:157]: It's not a question of whether `libraries.nix` needs `i18n` or not, we know `base.nix` needs it. It's a question of whether we should change the "API" of `kicad-unstable.passsthru`. Someone out there may be relying on it, if they have maybe some other build linking against `kicad-unstable.libraries.i18n`, this PR might tell them to use `passthru.i18n` instead, and I'm just unsure whether this is justified.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 13 above)
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/base.nix:54]: [I said before](https://github.com/NixOS/nixpkgs/pull/100066#issuecomment-706604693) that this PR triggers a change in the inputs of `kicad-unstable`, and it shouldn't. I said it shouldn't because it's only a change in the way the derivations are evaluated, and not in the derivations themselves. But, this is an example of a _justified_ "change" in one of the dependencies that triggers a rebuild, but isn't much of a big deal (only a change in the value of `version` which affects it's `name`). Hence I take back my words that this PR shouldn't trigger rebuilds - so there's nothing for you to do around this.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 14]
```diff
@@ -41,18 +41,17 @@
 , gtk3
 }:
 
-assert ngspiceSupport -> libngspice != null;
-
+# oce on aarch64 fails a test
+assert withOCE -> !stdenv.isAarch64;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:47]: Use `stdenv.lib.assertions.assertMsg`.

https://nixos.org/manual/nixpkgs/unstable/#sec-functions-library-asserts

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 15]
```diff
@@ -28,7 +31,13 @@
...
 }:
 
-assert ngspiceSupport -> libngspice != null;
+assert withNgspice -> libngspice != null;
+assert stdenv.lib.assertMsg (!ngspiceSupport)
+  "nspiceSupport is deprecated. Use withNgspice instead.";
+assert stdenv.lib.assertMsg (!oceSupport)
+  "oceSupport is deprecated. Use withOCE instead.";
+assert stdenv.lib.assertMsg (!scriptingSupport)
+  "scriptingSupport is deprecated. Use withScripting instead.";
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:77]: Excellent.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 16]
```diff
@@ -44,29 +46,16 @@
...
+  src = kicadSrc;
 
-  src = fetchFromGitLab (
-    {
-      group = "kicad";
-      owner = "code";
-      repo = "kicad";
-    } // versionConfig.kicadVersion.src
-  );
+  inherit i18n;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:58]: Not needed, as it's used in `postInstall`.

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 9 above)
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:157]: We are already doing some backwards incompatible changes. Hence I'd keep things as are :). Sorry for maybe misleading you. I'm thinking as I'm reviewing.

**Review by doronbehar — COMMENTED** [review]
Not many more to go. The commit messages look good overall. Only commit https://github.com/NixOS/nixpkgs/commit/239d16c2c20971984f91f083cce5be6f373e5280 should mention the rename of the `*Support` arguments to `with*`.

**Comment by matthuszagh** [comment]
I've implemented the changes. Let me know if there's anything else.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 17]
```diff
@@ -41,18 +41,16 @@
 , gtk3
 }:
 
-assert ngspiceSupport -> libngspice != null;
-
+assert stdenv.lib.asserts.assertMsg (!(withOCE && stdenv.isAarch64)) "OCE fails a test on Aarch64";
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:46]:
```suggestion
assert stdenv.lib.asserts.assertMsg (!(withOCE && stdenv.isAarch64)) "OCE fails a test on Aarch64";
# WHY?
assert stdenv.lib.asserts.assertMsg (withOCCT && !withOCE) "Can't enable OCCT and not OCE";
```

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 18]
```diff
@@ -41,18 +41,16 @@
...
 let
   versionConfig = versions.${baseName};
 
-  # oce on aarch64 fails a test
-  withOCE = oceSupport && !stdenv.isAarch64;
-  withOCC = (withOCCT && !withOCE) || (oceSupport && stdenv.isAarch64);
-
   libraries = callPackages ./libraries.nix versionConfig.libVersion;
 
+  withOCC = assert (withOCCT && !withOCE) || (withOCE && stdenv.isAarch64); true;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:48]: The idea was to remove this messy assignment. What is the problem with cross enabling all of these features? I understand the lack of aarch64 support, but not this one. Maybe the expression should be simpler in favor of a failing build?

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 19]
```diff
@@ -25,13 +25,13 @@
 , stable ? true
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:25]: The commit message should be (IMO):

```
kicad: Rename `<feature>Support` arguments to `with<Feature>`

Also: Use assertions instead of silently ignoring arguments that don't cooperate
(occt+oce) / won't compile (aarch64 + oce)
```

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 20]
```diff
@@ -25,13 +25,13 @@
...
 , baseName ? "kicad"
 , versions ? { }
-, oceSupport ? false
+, withOCE ? false
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:30]: Slight nitpick: The default values are already inherited from `default.nix` so it shouldn't be needed to specify them here as well, I think.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 18 above)
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/base.nix:48]: Also the use of both `withOCC` variable and `withOCCE` is confusing for someone reading this.

**Review by doronbehar — COMMENTED** [review]
Thanks for your cooperation. Only issues I see are in the assertions commit. It's almost perfect.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 20 above)
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/base.nix:30]: It would be preferred to put out the defaults in `default.nix` and not list them here as well, as again - `default.nix` is the 1st thing a 3rd party reader reads.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 18 above)
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/base.nix]: I've taken this statement out and added an assertion to ensure occ and oce are not both enabled. Kicad doesn't actually impose this requirement on us; it simply uses occ if occ and oce are both specified:

```
option( KICAD_USE_OCE
    "Build tools and plugins related to OpenCascade Community Edition (default ON)"
    ON )

option( KICAD_USE_OCC
    "Build tools and plugins related to OpenCascade Technology (overrides KICAD_USE_OCE, default OFF)"
    OFF )
```

So, if you think it's better I can take out the assertion.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 20 above)
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/base.nix]: I removed all defaults in addition to the `with<feature>` ones since they're all provided in default.nix. Let me know if that's too far and I'll add those back in.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 21]
```diff
@@ -61,7 +70,7 @@ stdenv.mkDerivation rec {
   base = callPackage ./base.nix {
     inherit versions stable baseName;
     inherit wxGTK python wxPython;
-    inherit debug withI18n withOCCT oceSupport ngspiceSupport scriptingSupport;
+    inherit debug withI18n withOCCT withOCE withNgspice withScripting;
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:166]:
```suggestion
    inherit debug withI18n withOCC withOCE withNgspice withScripting;
```

Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:166]: Or maybe use `withOCCT` on `base.nix` as well.

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 22]
```diff
@@ -22,37 +22,36 @@
...
-, baseName ? "kicad"
-, versions ? { }
-, oceSupport ? false
+, stable
+, baseName
+, versions
+, withOCE
 , opencascade
-, withOCCT ? true
+, withOCC
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/base.nix:32]: Same as it used to be.

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 21 above)
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/default.nix:166]: Ah right, forgot about default.nix. Isn't OCCT confusing here though since the kicad cmake parameter is OCC?

Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/default.nix:166]: Or, better to keep the interface the same, hence using OCCT?

Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:166]: Yea... It's up to you whether to change this API as well. I would. With a similar deprecation warning as the others.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 20 above)
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/base.nix]: No that's perfect.

File: pkgs/applications/science/electronics/kicad/base.nix (hunk 18 above)
Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/base.nix]: No keep it, besides the typo, everything is perfect IMO - this as well.

**Comment by matthuszagh** [comment]
I've made the changes, let me know what you think @doronbehar.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 23]
```diff
@@ -28,7 +32,15 @@
...
-assert ngspiceSupport -> libngspice != null;
+assert withNgspice -> libngspice != null;
+assert stdenv.lib.assertMsg (!ngspiceSupport)
+  "nspiceSupport is deprecated. Use withNgspice instead.";
+assert stdenv.lib.assertMsg (!oceSupport)
+  "oceSupport is deprecated. Use withOCE instead.";
+assert stdenv.lib.assertMsg (!scriptingSupport)
+  "scriptingSupport is deprecated. Use withScripting instead.";
+assert stdenv.lib.assertMsg (!withOCCT)
+  "withOCCT is deprecated. Use withOCC instead.";
```
Comment (doronbehar) [inline pkgs/applications/science/electronics/kicad/default.nix:80]: On a 2nd thought, now it hits me that we didn't have a deprecation period :laughing: . Personally I'm in favor of doing small incompatible changes like this without warning, but at least we should comply to the definition of the words we are using. So maybe (example):

```
Please use `withOCCT` instead of `withOCC`
```

Or:

```
`withOCCT` was renamed to `withOCC` for the sake of consistency with upstream cmake options
```

And:

```
`scriptingSupport` was renamed to `withScripting` for the sake of consistency with other kicad nix arguments.
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
Also, to be nice to everyone else, we should add release notes to rl-2103.xml (ideally in seperate commit from this one).

The commit message of this "rename options commit" should mention also the fact that the defaults are not needed in `base.nix`, although it doesn't affect users directly.

1 last round I promise!

File: pkgs/applications/science/electronics/kicad/default.nix (hunk 23 above)
Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/default.nix:80]: You mean take out these deprecation assertions? I'm happy to do that if you think it's better. If we leave them in, how long before I should come back and take them all out?

And don't worry about the many rounds of edits, better to get this right the first time than to have to come back and fix it later :)

Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:80]: I meant only to change the message.

Comment (doronbehar) [reply pkgs/applications/science/electronics/kicad/default.nix:80]: Please leave the assertions as are.

Comment (matthuszagh) [reply pkgs/applications/science/electronics/kicad/default.nix]: Got it. Ignore that last force push. I still need to add the release notes.

**Comment by doronbehar** [comment]
I know it ruins the `git commit --fixup=` workflow, but could you please mention in that commit the fact that `base.nix` doesn't need the defaults to be specified in it?

Also, we'll definitely need a commit that mentions these api changes in `rl-2103.xml`.

**Comment by matthuszagh** [comment]
Changes in.

**Review by doronbehar — APPROVED** [review]
Thanks for your contribution.

File: pkgs/applications/science/electronics/kicad/default.nix [hunk 24]
```diff
@@ -1,68 +1,192 @@
...
+    group = "kicad";
+    owner = "code";
+    repo = "kicad-i18n";
+    rev = versionsImport.${baseName}.libVersion.libSources.i18n.rev;
+    sha256 = versionsImport.${baseName}.libVersion.libSources.i18n.sha256;
+  };
 
-  versions =  import ./versions.nix;
-  versionConfig = versions.${baseName};
+  libSrcFetch = name: fetchFromGitLab {
```
Comment (evils) [inline pkgs/applications/science/electronics/kicad/default.nix:103]: one of the goals of my rewrite was to simplify default.nix to be as much as possible just the wrapper
code that's related to building the components that get tied together here should be in base.nix or libraries.nix respectively

File: pkgs/applications/science/electronics/kicad/libraries.nix [hunk 25]
```diff
@@ -42,30 +28,4 @@ in
   templates = mkLib "templates";
   footprints = mkLib "footprints";
   packages3d = mkLib "packages3d";
-
-  # i18n is a special case, not actually a library
```
Comment (evils) [inline pkgs/applications/science/electronics/kicad/libraries.nix:46]: i think this description of i18n being optional, separate and only found via a relative path should have been retained

**Review by evils — COMMENTED** [review]
sorry for such a late review
i missed it because i confused its mails for #85456
and it being merged just 3 days after being opened did not leave a lot of time to discover that mistake
